### PR TITLE
Add Webpack to collect and Serve static files

### PR DESCRIPTION
**Pull request by joaoluizn**
To summarize this commit:

- I've configured Django to render a single index.html page that will be dynamically changed by the Vue.js collected from _webpack_.

- **_webpack:dev:server_** is the command used to run webpack server, collect static files and serve then, keeping it on memory.

- **_webpack:dev:watch_** is used to generate bundles files instead of keeping then on memory.

that is all folks!

### CODE REVIEW

File: frontend/public/index.html [hunk 1]
```diff
@@ -1,19 +1,27 @@
...
-    <title><%= htmlWebpackPlugin.options.title %></title>
-  </head>
-  <body>
-    <noscript>
-      <strong>We're sorry but <%= htmlWebpackPlugin.options.title %> doesn't work properly without JavaScript enabled. Please enable it to continue.</strong>
-    </noscript>
-    <div id="app"></div>
-    <!-- built files will be auto injected -->
-  </body>
+    <title>Django Vue Integration</title>
```
Comment (marlom-jobsom) [inline frontend/public/index.html]: Can we keep it the previous title markup?
`<title><%= htmlWebpackPlugin.options.title %></title>`

File: frontend/src/App.vue [hunk 2]
```diff
@@ -1,28 +1,28 @@
 <template>
-  <div id="app">
-    <img alt="Vue logo" src="./assets/logo.png">
-    <HelloWorld msg="Welcome to Your Vue.js App"/>
-  </div>
+    <div id="app">
+        <img alt="Vue logo" :src="require('../assets/img/logo.png')">
```
Comment (marlom-jobsom) [inline frontend/src/App.vue]: How about to use the complete path to the asset `logo.png`?

File: frontend/src/App.vue [hunk 3]
```diff
@@ -1,28 +1,28 @@
...
 
-export default {
-  name: 'App',
-  components: {
-    HelloWorld
-  }
-}
+    export default {
+        name: 'App',
+        components: {
```
Comment (marlom-jobsom) [inline frontend/src/App.vue]: Correct the identation with 4 spaces only

File: frontend/src/main.js [hunk 4]
```diff
@@ -1,10 +1,10 @@
 import Vue from 'vue'
 import App from './App.vue'
-import vuetify from '@/plugins/vuetify'
+import vuetify from './plugins/vuetify'
```
Comment (marlom-jobsom) [inline frontend/src/main.js]: What is the difference of use `@`?

File: pycdg/settings.py [hunk 5]
```diff
@@ -120,8 +117,19 @@
...
+
+STATICFILES_DIRS = (
+    os.path.join(FRONTEND_DIR, 'assets'),
+)
+
+# Webpack loader setup
+WEBPACK_LOADER = {
+    'DEFAULT': {
+        'CACHE': DEBUG,
+        'BUNDLE_DIR_NAME': os.path.join(FRONTEND_DIR, '/bundles/'),  # must end with slash
```
Comment (marlom-jobsom) [inline pycdg/settings.py]: The `os.path.join()` already add the director separator for us

File: pycdg/urls.py [hunk 6]
```diff
@@ -18,10 +18,15 @@
...
 from django.urls import include
 from django.urls import path
+from django.views.generic import TemplateView
 
 from teammate.routers import router
 
 urlpatterns = [
     url(r'api/', include(router.urls)),
     path('admin/', admin.site.urls),
+    path('',
```
Comment (marlom-jobsom) [inline pycdg/urls.py]: Keep that inline

File: frontend/src/App.vue (hunk 3 above)
Comment (joaoluizn) [reply frontend/src/App.vue]: Its already using 4 spaces, its the github preview that is weird, try to select from the green mark up to the + symbol

File: frontend/public/index.html (hunk 1 above)
Comment (joaoluizn) [reply frontend/public/index.html]: Done

File: pycdg/urls.py (hunk 6 above)
Comment (joaoluizn) [reply pycdg/urls.py]: Done

File: pycdg/settings.py (hunk 5 above)
Comment (joaoluizn) [reply pycdg/settings.py]: Done

File: frontend/src/main.js (hunk 4 above)
Comment (joaoluizn) [reply frontend/src/main.js]: Done

File: README.md [hunk 7]
```diff
@@ -3,11 +3,11 @@
 Coding Data Gatherer (CDG) aims to **store data** regarding the effort of a developers team when 
 working on softwares projects to **generate views** to make easy to follow-up the progress.
 
-# Requirements
+## Requirements
```
Comment (marlom-jobsom) [inline README.md]: It's not a sub-topic from the presentation pyCDG

File: README.md [hunk 8]
```diff
@@ -3,11 +3,11 @@
...
 * [Pipenv](https://pipenv.kennethreitz.org/en/latest/)
 * [Yarn](https://classic.yarnpkg.com/en/docs/install)
 
-# Environment Setup
+## Environment Setup
```
Comment (marlom-jobsom) [inline README.md]: It's not a sub-topic from the presentation pyCDG

File: README.md [hunk 9]
```diff
@@ -17,13 +17,13 @@ cd frontend
 yarn install
 ```
 
-# Run with hot-reloads for development
+## Run for development with hot reload
```
Comment (marlom-jobsom) [inline README.md]: 1 - It's not a sub-topic from the presentation pyCDG
2 - hot-reloads

File: frontend/src/main.js [hunk 10]
```diff
@@ -2,9 +2,9 @@ import Vue from 'vue'
 import App from './App.vue'
 import vuetify from '@/plugins/vuetify'
 
-Vue.config.productionTip = false
+Vue.config.productionTip = false;
```
Comment (brunohgv) [inline frontend/src/main.js]: remove semicolon

File: frontend/src/main.js [hunk 11]
```diff
@@ -2,9 +2,9 @@ import Vue from 'vue'
...
 
 new Vue({
-  vuetify,
-  render: h => h(App),
-}).$mount('#app')
+    vuetify,
+    render: h => h(App),
+}).$mount('#app');
```
Comment (brunohgv) [inline frontend/src/main.js]: remove semicolon

File: pycdg/settings.py [hunk 12]
```diff
@@ -120,8 +117,19 @@
...
+
+STATICFILES_DIRS = (
+    os.path.join(FRONTEND_DIR,  *['src', 'assets']),
+)
+
+# Webpack loader setup
+WEBPACK_LOADER = {
+    'DEFAULT': {
+        'CACHE': DEBUG,
+        'BUNDLE_DIR_NAME': os.path.join(FRONTEND_DIR, 'bundles'),  # must end with slash
```
Comment (marlom-jobsom) [inline pycdg/settings.py]: What is that comment for?

File: frontend/webpack.config.js [hunk 13]
```diff
@@ -1,30 +1,145 @@
-// webpack.config.js
+const path = require('path');
+const webpack = require('webpack');
+const MiniCssExtractPlugin = require('mini-css-extract-plugin');
+const BundleTracker = require('webpack-bundle-tracker');
+const VuetifyLoaderPlugin = require('vuetify-loader/lib/plugin');
+const {VueLoaderPlugin} = require('vue-loader');
```
Comment (brunohgv) [inline frontend/webpack.config.js]: Remove semicolons

**Pull request merged**